src/persist/src/location.rs

```diff
@@ -247,11 +286,11 @@ pub trait Consensus: std::fmt::Debug {
     /// This data is initialized to None, and the first call to compare_and_set needs to
     /// happen with None as the expected value to set the state.
     async fn compare_and_set(
-        &mut self,
+        &self,
```

Comment:
needed this to make the `Arc<dyn Consensus>` pattern work. I'm still unsure where I come down on this. it's logically a mutation, but OTOH`&mut` is tied to rust's safety reasoning and `compare_and_set` is definitely safe to  call concurrently